tests/test_external_mixin.py

```diff
@@ -377,3 +421,146 @@ def test_restore_state(self, strategy):
         assert strategy._connected is False
         assert strategy._use_template_strategy is False
         strategy.state.restore_state.assert_called_once_with(state_dict)
+
+
+class TestForecastRelatedFeatures:
+    @pytest.mark.parametrize("ext_strategy_fixture", [LoadForecastExternalStrategy(),
+                                                      PVForecastExternalStrategy()], indirect=True)
+    def test_set_energy_forecast(self, ext_strategy_fixture):
```

Comment:
Nit again, but this test could be split into 2 IMO: test_set_energy_forecast_command_is_accepted and test_energy_forecast_command_fails_wrong_arguments . In addition to that, the first test could also include the check that the publish_json method has been called (because even on success we still respond via Redis IIRC).